### PR TITLE
Eager query method & Clean options for eager loading.

### PR DESCRIPTION
**Pull request by larrymjordan**
Hi @markbates this solves issue https://github.com/gobuffalo/pop/issues/72

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -182,6 +182,12 @@ func (c *Connection) Q() *Query {
 	return Q(c)
 }
 
+// eagerDisabled disables eager mode for current connection.
+func (c *Connection) eagerDisabled() {
```
Comment (paganotoni) [inline connection.go]: @larrymjordan should this be called something like "disableEager" ?

Comment (larrymjordan) [reply connection.go]: @paganotoni yes, that makes sense.

**Comment by markbates** [comment]
@stanislas-m @larrymjordan I'm also concerned about storing "state" information on the connection. How do we solve this problem? Is this PR safe to merge, while a solution is found? or should this PR wait? You both know more on the subject than I do so I'm going to trust you two. :)

**Comment by markbates** [comment]
My first reaction was copying transaction too. But I think returning a query is better, but it’s a breaking change. 

I think what someone really needs to do is write up an issue describing how this feature should work, including different use cases, etc…

**Comment by larrymjordan** [comment]
@markbates @stanislas-m The reason why `Eager` function is attached to `Connection` is because of the executors `Create`, `ValidateAndCreate` methods. The only way to access these methods is through Connection object. So the way I handled that to be able to introduce `Eager` creation was by adding those states to `Connection`.  By returning a query with eager how can we use it for eager Creation? Any ideas?

**Comment by stanislas-m** [comment]
@larrymjordan As I proposed, I think for `Create` and `ValidateAndCreate` we can create a copy of the Connection. This way, the eager feature is isolated from other requests.

**Comment by larrymjordan** [comment]
@stanislas-m can you take a look when you have the chance to my last commit, is that what you suggested?

**Comment by stanislas-m** [comment]
@larrymjordan Sure, I'll take a look in a few minutes.

**Comment by markbates** [comment]
@larrymjordan @stanislas-m great work!!

**Comment by trevrosen** [comment]
@larrymjordan @stanislas-m just ran into the underlying bug -- thanks so much for taking the time to produce a fix!

**Pull request merged**